### DISCUSSION TITLE
User Preferences - Communications

### DISCUSSION THREAD

**Discussion by tnurse18** [WPI-LNL/lnldb#565] [Ideas]
In the coming weeks we will be working on introducing individual user preferences for the LNLDB. One piece of this will involve presenting options for selecting which kinds of messages a user would like to receive from LNL and through what methods they would like to receive them. What are everyone’s thoughts on the breakdown of some of the options listed below?

- Crew Chief Add Notifications
    - Email (Optional)
    - Slack (Optional)
- Crew Chief Report Reminders
    - Email (Optional or Required?)
    - Slack (Optional)
    - _Note: Users may not opt out completely. Either email is required and Slack is available as an additional option, or at least one of these will need to be checked. Not sure which approach would be better…thoughts?_
- Crew Chief Needed Notifications
    - Email and Slack both required
- Event Edited Notifications
    - Email (Optional)
    - Slack (Optional)
- Web Service Notifications
    - Email, Slack, and SMS (if applicable) are all required

Do you agree with these options? Can you think of any other types of communications we should add to this and if so, what options should be presented to users (i.e. email, SMS, slack, etc.)?

**Comment by Muirrum** [comment]
Can you elaborate more on "web service notifications"? What sort of thing would you send under that category?

**Reply by tnurse18** [reply]
This includes automatic account-related messages such as password reset emails (which for obvious reasons you can't unsubscribe from) as well as notices that I might send out to inform users of events such as service outages or changes to our data and privacy policies. I suppose this could also refer to some of the notifications you might receive from Snipe or RT (such as when you submit a ticket).

**Comment by kylemikableh** [comment]
I like that you can disable Event Edited notifications, but could you still require emails/slack for specific fields edited? Like name change or description isn't that important, but if the date is changed or status or something I feel like that should be required, especially for the VP or HP. Looks great otherwise!

**Reply by Muirrum** [reply]
+1 to this

(as an aside if the notification emails could tell you *what* got edited and the new/old values that'd be super helpful)

**Reply by tnurse18** [reply]
I think we might be able to do that. Do you have any suggestions on which fields should be required and/or for who?

**Reply by kylemikableh** [reply]
Anything related to datetime (e.g. datetime_end), location, services requested (not sure what this one is called) should be required. Possibly internal_notes too as usually something important goes there.

Anything related to fields such as description, send_survey, event_name, billing_org, billed_in_bulk should be able to be ignored as these really don't effect anything. If it does not contain what was edited, I think that means a minor change and ignore that?

**Reply by tnurse18** [reply]
Here's a breakdown of what I have at the moment for fields that could trigger a notification. Users would receive notifications for changes to required fields (no ability to opt out) and have the ability to opt-in to notifications for changes to the optional fields. Thoughts?

- Event name (Optional)
- Description (Optional)
- Location (Required)
- Contact (Optional)
- Billing org (Optional)
- Datetime setup complete (Required)
- Datetime start (Required)
- Datetime end (Required)
- Internal notes (Optional)
- Billed in bulk (Optional)
- Client (Optional)

**Comment by otherNatalie** [comment]
Do you want to add meeting announcements to this list? Or that might be more complicated (and lead to some bylaws questions) but I know that’s something some people would probably appreciate

**Reply by tnurse18** [reply]
Ideally, yes. But because those go to an alias, that becomes significantly more challenging. Users would need to leave the alias to opt-out. Since these usually go to lnl-news, that would mean opting out of work announcements and other things as well. I could probably mention this on the preferences page, but the LNLDB would have no control over it unless we changed the way we send them. And yes, that would likely mean bylaw changes.
